### PR TITLE
#366 Simplify GitHub Actions build workflow

### PR DESCRIPTION
**Pull request by yruslan**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: "adopt@1.8"
+      - uses: actions/checkout@v2
+      - uses: coursier/cache-action@v5
+      - name: Build and run tests
+        run: sbt ++${{matrix.scala}} test -DSPARK_VERSION=${{matrix.spark}}
+
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (Zejnilovic) [inline .github/workflows/build.yml:34]: Not sure what this is. Leftover?

Comment (yruslan) [reply .github/workflows/build.yml:34]: Yes, good spot, Thanks!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            spark: 3.0.1
+    name: Spark ${{matrix.spark}} on Scala ${{matrix.scala}}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: "adopt@1.8"
+      - uses: actions/checkout@v2
```
Comment (Zejnilovic) [inline .github/workflows/build.yml:28]: this line is extra

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: "adopt@1.8"
+      - uses: coursier/cache-action@v5
+      - name: Build and run tests
+        run: sbt ++${{matrix.scala}} test -DSPARK_VERSION=${{matrix.spark}}
```
Comment (yruslan) [inline .github/workflows/build.yml:30]: Now that we have to cache I'm wondering if it makes more sense to use 
```suggestion
        run: sbt clean ++${{matrix.scala}} test -DSPARK_VERSION=${{matrix.spark}}
```
?

Comment (yruslan) [reply .github/workflows/build.yml:30]: Not sure. It caches downloaded sbt files only (.sbt), right?

**Pull request merged**